rust/connlib/tunnel/src/peer/nat_table.rs

```diff
@@ -39,6 +39,11 @@ impl NatTable {
         }
     }
 
+    /// Returns true if the NAT table has any entries with the given "inside" IP address.
+    pub(crate) fn has_entry_for_inside(&self, ip: IpAddr) -> bool {
+        self.table.left_values().any(|(_, c)| c == &ip)
+    }
```

Comment:
This is `O(N)` but it shouldn't happen very often because we only use it whenever a resolve a DNS query (and it is local to a particular connection of a client which usually doesn't have that many NAT sessions open).